homeassistant/components/fritz/common.py

```diff
@@ -230,11 +234,18 @@ async def async_setup(
 
     def setup(self) -> None:
         """Set up FritzboxTools class."""
+
+        if self.port is None:
+            port = DEFAULT_HTTPS_PORT if self.use_tls else DEFAULT_HTTP_PORT
+        else:
+            port = self.port
+
```

Comment:
this should be handled in the config flow, so that a configuration entry always has a valid port in its data